src/com/google/javascript/jscomp/GlobalNamespace.java

```diff
@@ -910,6 +910,7 @@ Name getOrCreateName(String name) {
     private boolean hasDeclaredTypeDescendant = false;
```

Comment:
Wrong field changed. Change the declaredType one instead. (I find it confusing that it's called declaredType when its type is boolean.)